Tests/ApolloTests/RequestCreatorTests.swift

```diff
@@ -283,4 +284,135 @@ Bravo file content.
+
+    XCTAssertEqual(query.queryDocument, req["query"] as? String)
+  }
+
+  // MARK: - Custom request creator tests
+
+  func testSingleFileWithCustomRequestCreator() throws {
+    let alphaFileUrl = self.fileURLForFile(named: "a", extension: "txt")
+
+    let alphaFile = GraphQLFile(fieldName: "upload",
+                                originalName: "a.txt",
+                                mimeType: "text/plain",
+                                fileURL: alphaFileUrl)
+
+    let data = try customRequestCreator.requestMultipartFormData(
+      for: HeroNameQuery(),
+      files: [alphaFile!],
+      sendOperationIdentifiers: false,
+      serializationFormat: JSONSerializationFormat.self,
+      manualBoundary: "TEST.BOUNDARY"
+    )
+
+    let stringToCompare = try self.string(from: data)
+
+    print(stringToCompare)
+
+    if #available(iOS 11, macOS 13, tvOS 11, watchOS 4, *) {
+      let expectedString = """
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_variables"
+
+{"episode":null}
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_query"
+
+query HeroName($episode: Episode) { hero(episode: $episode) { __typename name } }
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="test_operationName"
+
+HeroName
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="upload"; filename="a.txt"
+Content-Type: text/plain
+
+Alpha file content.
+
+--TEST.BOUNDARY--
+"""
+      XCTAssertEqual(stringToCompare, expectedString)
+    } else {
+      // Operation parameters may be in weird order, so let's at least check that the files and single parameter got encoded properly.
+      let expectedEndString = """
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="map"
+
+{"0":["variables.upload"]}
+--TEST.BOUNDARY
+Content-Disposition: form-data; name="0"; filename="a.txt"
+Content-Type: text/plain
+
+Alpha file content.
+
+--TEST.BOUNDARY--
+"""
+      self.checkString(stringToCompare, includes: expectedEndString)
+    }
+  }
+
+  func testRequestBodyWithCustomRequestCreator() {
+    let query = HeroNameQuery()
+    let req = customRequestCreator.requestBody(for: query, sendOperationIdentifiers: false)
+
+    XCTAssertEqual(query.queryDocument, req["test_query"] as? String)
+  }
+}
+
+fileprivate struct TestRequestCreator: RequestCreator {
+  public func requestBody<Operation: GraphQLOperation>(for operation: Operation, sendOperationIdentifiers: Bool) -> GraphQLMap {
+    var body: GraphQLMap = [
+      "test_variables": operation.variables,
+      "test_operationName": operation.operationName,
+    ]
+
+    if sendOperationIdentifiers {
+      guard let operationIdentifier = operation.operationIdentifier else {
+        preconditionFailure("To send operation identifiers, Apollo types must be generated with operationIdentifiers")
+      }
+
+      body["test_id"] = operationIdentifier
+    } else {
+      body["test_query"] = operation.queryDocument
+    }
+
+    return body
```

Comment:
Are this implementation and the one below substantially different from the real implementation? Might be worth making them more obviously different to help make sure the "correct" implementation is being used.

Reply:
The multipart is the one before the Apollo "standard" was added. It is actually the one that works with Absinthe GraphQL 😅

I have changed the names of the keys. 
I think it is okay as we only want to ensure the right method is called?
So all keys in the `requestBody` starts with `test_`. I think that is good enough to ensure that we not call the default implementation